lib/active_record/virtual_attributes/virtual_total.rb

```diff
@@ -77,20 +96,19 @@ def define_virtual_aggregate_attribute(name, relation, method_name, column, opti
         end
       end
 
-      def define_virtual_size_method(name, relation)
-        define_method(name) do
-          (has_attribute?(name) ? self[name] : send(relation).try(:size)) || 0
-        end
-      end
-
-      def define_virtual_aggregate_method(name, relation, method_name, column)
+      def define_virtual_aggregate_method(name, relation, column, ruby_method_name, arel_method_name = ruby_method_name)
         define_method(name) do
           if has_attribute?(name)
             self[name] || 0
           elsif (rel = send(relation)).loaded?
-            rel.map { |t| t.send(column) }.compact.send(method_name)
+            values = rel.map { |t| t.send(column) }.compact
+            if block_given?
+              yield values
+            else
+              rel.blank? ? nil : values.send(ruby_method_name)
```

Comment:
I think it should look at values maybe?
```suggestion
              values.blank? ? nil : values.send(ruby_method_name)
```